### PR TITLE
Update to API version 2.2

### PR DESCRIPTION
**Pull request by bhelx**
Updating to API version 2.2

I had to add support for `updated_at` in the appropriate places. I'd like to also add usage based billing but I think that might take some more time and I want to assess the demand. This is good enough to unblock us and get us to version 2.3.

### CODE REVIEW

File: Library/AddOn.cs [hunk 1]
```diff
@@ -6,11 +6,16 @@ namespace Recurly
...
-
         public string PlanCode { get; set; }
         public string AddOnCode { get; set; }
         public string Name { get; set; }
-
+        public int DefaultQuantity { get; set; }
+        public bool? DisplayQuantityOnHostedPage { get; set; }
+        public string TaxCode { get; set; }
+        public bool? Optional { get; set; }
+        public String AccountingCode { get; set; }
```
Comment (csmb) [inline Library/AddOn.cs]: Should we add `xmlWriter.WriteElementString("accounting_code", AccountingCode);` [further down](https://github.com/recurly/recurly-client-net/pull/154/files#diff-e84c4a9c2623e09a15d1c2b70fecc921R164)?

Comment (bhelx) [inline Library/AddOn.cs]: Yes, appears I missed it. Good eye.

Comment (MuffinMan10110) [inline Library/AddOn.cs]: @csmb Looks like the API accepts setting that value when creating a plan add-on. If you don't set it then it gets set to the value of `AddOnCode`. So it should be a valid field to write to the request xml.

**Comment by bhelx** [comment]
@csmb okay updated the reader and writer. Sorry it took so long.

**Comment by csmb** [comment]
+1

**Pull request merged**